### PR TITLE
chore: Upgrade ocular-dev-tools to tsc+esbuild

### PR DESCRIPTION
**Pull request by ibgreen**
This upgrade brings a massive 10x performance increase for our builds.

@belom88 @dsavinov-actionengine 
- I suspect this may break publishing for a little while until we fix any issues. 
- I'd like to merge it as soon as possible - but want to make sure I don't block your work.

### CODE REVIEW

**Comment by Pessimistress** [comment]
Suggestions for testing local build:

- run `yarn bootstrap`
- in CLI, enter node and inspect:
  + `import('@loaders.gl/core').then(console.log)`
  + `require('@loaders.gl/core')`
- go to `modules/core`, run `yarn build-bundle`. Create a temporary html file with `<script src="dist.min.js"></script>`, open in browser and inspect the global `loaders` object.

**Comment by ibgreen** [comment]
> Suggestions for testing local build:

Both these tests fail

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "start": "echo 'Please see loaders.gl website for how to run examples' && open https://loaders.gl/docs",
     "clean": "ocular-clean",
     "tsclean": "find . -name tsconfig.tsbuildinfo -exec rm {} \\;",
-    "build": "ocular-build && lerna run pre-build",
+    "build": "ocular-clean && ocular-clean && ocular-build && lerna run pre-build",
```
Comment (ibgreen) [inline package.json:32]: That is a lot of cleans... Do we need to clean everytime we build?

File: modules/arrow/package.json [hunk 2]
```diff
@@ -44,8 +44,9 @@
     "fs": false
   },
   "scripts": {
-    "pre-build": "npm run build-worker && npm run build-bundle && npm run build-bundle -- --env=dev && npm run build-triangulation-worker",
-    "build-bundle": "ocular-bundle ./src/index.ts",
+    "pre-build": "npm run build-worker && npm run build-bundle && npm run build-bundle-dev && npm run build-triangulation-worker",
```
Comment (ibgreen) [inline modules/arrow/package.json:47]: Should this be `prepublishOnly` everywhere? Or good to stick with `pre-build`? Efficient, but a bit uncanny that some builds only happen on publish.

Comment (Pessimistress) [reply modules/arrow/package.json:47]: Some packages require pre-build to run (copy of libs, build workers etc.)

**Pull request merged**